### DISCUSSION TITLE
Problem with document links

### DISCUSSION THREAD

**Discussion by ZimTom** [zim-desktop-wiki/zim-desktop-wiki#2193] [General]
In my ZIM-Wiki (Version 0.75 with Win10 Pro x64) I regularly use the standard function of having files of all kinds attached to a Wiki page by storing them in the attachment directory created by ZIM and everything worked fine and flawlessly.
Now, for some reason I have problems to continue doing that and even in some of the existing pages links that used to work I cannot open attached documents by clicking on a link anymore. Instead of the document being opened I get the message "Das System kann die angegebene Datei nicht finden ...", i.e. file not found in the language selected for the ZIM installation, although
- the check of the TXT files shows that everything is o.k.,
- the required directory exists and
- the required file file is there with the correct name.

If I delibrately delete the linked file or change its name, clicking the link to it leads to the message "No such file or folder: ..." directly in English ignoring the installation language. This shows that ZIM recognizes the diference between a simply corrupted link (wrong file name of path) and the situation where a correct link cannot open the linked file.

Looking at the directory structure and names as well as the syntax in the TXT files I cannot find any systematic difference between pages where the links are working and those where they don't work.

I have tried the check & update index and even to reinstall ZIM, but up to now I can't get it working as it used to.

Does anybody have an idea what could cause this kind of behaviour ?

Thanks in advance for your help.

**Comment by jaap-karssenberg** [comment]
You are probably affected by this issue: 0.75 file links no longer work
(possibly because of accents in paths) · Issue #2168 ·
zim-desktop-wiki/zim-desktop-wiki · GitHub
<https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2168>

**Reply by ZimTom** [reply]
Thanks for the quick response. After reading this issue, I checked immediately, but accents can't be the problem here because I am working in German and English only and I also especially avoid special characters in file names as a habit - including the "German Umlauts" (äöüÄÖÜß).

- If I delete the not working link, change away from the page and back to make the modification become effective, and then create a new link, it doens't work, independent from whether I create via drag and drop or manually via menu option.
- If I insert a link in the same page, but to another, randomly chosen existing document, the link doesn't work.
- If I insert a link in the same page to a document that has a link to it in another ZIM page which is working, the new link in the test page is working as well.
- If I create a new page and include a link to the affected document the link doesn't work.
- If I create a new page and include a link to a document that has a link to it in another ZIM page which is working, the new link works  

And there there is something else I discovered in relation to this problem:

Normally a right click on a file link opens the dialog for editing, removing of a link etc.
I haven't found the pattern behind it, but I have seen situations where
- ZIM was started,
- it openend with the page wth the malfunctioning link and
- if I right click on the link, ZIM just closes without a comment after 1-2 seconds.

I can't prove it and I don't know whether it makes sense to you, but to me it seems like it could have something to do with the ability to open the corresponding app for visualizing the document:

- If I have a working link and deliberately corrupt it either by moving or renaming the target document or manipulating the link itself, I get the English "File not found" message and if I undo the manipulation the link works correctly again immediately, i.e. open the corresponding app and show the documemt content
- If I have a non-working link and deliberately corrupt it either by moving or renaming the target document or manipulating the link itself, I get the English "File not found" message and if I undo the manipulation, clicking the link generates the - in my cace - German message "Das System kann die angegebene Datei nicht finden: file:///..."

A link that has been proven to be correct in syntax and including the destination path to a valid document, opens the app in one case and in the other case not, although the document type (e.g. xlsx) is identical in both cases.

Wow, this is weird ...

**Comment by jaap-karssenberg** [comment]
Could you give an example of a link that does not work (please use "view
source") and the corresponding file name ?

Also a look at the debug log ("view" menu) would be helpfull.

--- Jaap

**Reply by ZimTom** [reply]
Here you are, attached you find
- a screenshot of the ZIM page where I first discovered the problem with the non-working link including the attachment directory underneath. Both links to the files shown at the bottom created via drag & drop don't work.
- a screenshot of the source of the page
- a copy of zim.log taken at a stage that should show the information required:
   - ZIM was started, then a right click on a link in the page caused ZIM to crash (closing without comment)
   - ZIM was started again and then one of the non-working links was clicked causing the the error message to pop up.

I don't know whether this means anything, but I realized that when using the "View Debug log" first the log wasn't opened because ZIM thinks that no application is defined for plain/text, although in Win I have associated Notepad++ with TXT files, so that the default setting to sysem standard should work. After adding Notepad++ directly everything worked.
There is no visible difference in source text between the example page with the non-workling link and other pages with working links to XLS>, PDF or other documents.

Hope the attached information is what you were asking for.

Best regards
 
 
![Non-Working_Link_01_2022-10-15 134225](https://user-images.githubusercontent.com/108861992/195986656-ef90558b-169c-4dde-9e11-dfe16f6ad3f2.jpg)
![Non-Working_Link_02_2022-10-15 134527](https://user-images.githubusercontent.com/108861992/195986658-8e4eacec-5a20-4f53-bf06-412ee6a84f8a.jpg)
[Non-Working_Link_03_2022-10-15 135113.txt](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/9792050/Non-Working_Link_03_2022-10-15.135113.txt)

**Comment by jaap-karssenberg** [comment]
Yes thank you, that helps a lot.

I still believe it is the same issue as referred above. The file itself
does not have any "German" letters, but there is an umlaut a parent folder.
Also I just realized usage of an "&" triggers the same issue.

I'm quite confident this will be fixed with the fix release next week.

Kind regards,

Jaap

Op za 15 okt. 2022 14:44 schreef ZimTom ***@***.***>:

**Reply by ZimTom** [reply]
Thanks a lot for your feedback. Glad to be able to help a little bit. Maybe I overlooked some of the obstacle characters and I will have a closer look around, try to get rid of them and see whether this will cure the problem with existing TXT files and containing links even before the update is available and I am happy that hopefully pretty soon I will be able to use ZIM on a Windows machine AND the first Mac of my life :-)

Best regards and a nice Sunday to you

Thomas

**Reply by ZimTom** [reply]
O.k. I have gone through my ZIM tree and replaced all umlaut's in document headers by renaming the pages and without touching the content of the pages all links are working again as normal.
That said, in my case I could see that links sitting somwhere in a document in a path containing a "&" in file or directory names, but not containing an umlaut were still working. But after your comment I replaced all "&" with "+" to be on the safe side.

With the upcoming update I can then gradually re-introduce these "beloved traditional characters that get in the way of software in so many occasions :-D

Best regards
Thomas
